### ISSUE TITLE
Upgrade ipmitool-xcat to 1.8.18 version

### ISSUE THREAD

**Issue by zet809** [xcat2/xcat-core#3194]
- Verify the signal handler patch for SOL on OpenPOWER machine.
- Build ipmitool-xcat 1.8.18 for the os+arch below:
  rh7 x86_64
  rh7 ppc64
  rh7 ppc64le
  rh6 ppc64
  rh6 x86_64

  ubuntu 14.04 x86_64
  ubuntu 14.04 ppc64le

  sles 11.4 x86_64 
  sles12 x86_64
  sles12 ppc64le
  sles11.4  ppc64

Issue #3213 cover the verification of the build.

**Comment by whowutwut** [comment]
@chenglch  i'll take this issue.

**Comment by zet809** [comment]
Hi, @whowutwut , maybe we only need to focus on RH7 ppc64, x86_64 and ppc64le, ubuntu 14.04 x86_64 and ppc64le(need to makesure the build on 14.04 can work on 16.04 or higher), and sles12 ppc64le and x86_64. For other arch, we can put it as low priority.

**Comment by whowutwut** [comment]
I've put the following files in GSA staging area: 

### Linux
```
[root@c910f02c07p02 xcat-dep]# pwd
/gsa/pokgsa/projects/x/xcat/build/linux/xcat-dep
[root@c910f02c07p02 xcat-dep]# find ./ -name "ipmitool*1.8.18*" -print 
./sles12/x86_64/ipmitool-xcat-1.8.18-0.x86_64.rpm
./sles12/ppc64le/ipmitool-xcat-1.8.18-0.ppc64le.rpm
./rh7/ppc64/ipmitool-xcat-1.8.18-0.ppc64.rpm
./rh7/x86_64/ipmitool-xcat-1.8.18-0.x86_64.rpm
./rh7/ppc64le/ipmitool-xcat-1.8.18-0.ppc64le.rpm
[root@c910f02c07p02 xcat-dep]# find ./ -name "ipmitool*1.8.18*" -print  | xargs sum
37011   288 ./sles12/x86_64/ipmitool-xcat-1.8.18-0.x86_64.rpm
32371   269 ./sles12/ppc64le/ipmitool-xcat-1.8.18-0.ppc64le.rpm
25033   295 ./rh7/ppc64/ipmitool-xcat-1.8.18-0.ppc64.rpm
43156   308 ./rh7/x86_64/ipmitool-xcat-1.8.18-0.x86_64.rpm
33101   294 ./rh7/ppc64le/ipmitool-xcat-1.8.18-0.ppc64le.rpm
```

And removing the corresponding 1.8.17-x version: 
```
[root@c910f02c07p02 xcat-dep]# rm -f ./sles12/x86_64/ipmitool-xcat-1.8.17-1.x86_64.rpm 
[root@c910f02c07p02 xcat-dep]# rm -f ./sles12/ppc64le/ipmitool-xcat-1.8.17-1.ppc64le.rpm 
[root@c910f02c07p02 xcat-dep]# rm -f ./rh7/ppc64/ipmitool-xcat-1.8.17-1.ppc64.rpm 
[root@c910f02c07p02 xcat-dep]# rm -f ./rh7/x86_64/ipmitool-xcat-1.8.17-1.x86_64.rpm 
[root@c910f02c07p02 xcat-dep]# rm -f ./rh7/ppc64le/ipmitool-xcat-1.8.17-1.ppc64le.rpm
```


### Debian 
```
[root@c910f02c07p02 xcat-dep]# pwd
/gsa/pokgsa/projects/x/xcat/build/ubuntu/xcat-dep
[root@c910f02c07p02 xcat-dep]# find ./ -name ipmitool*1.8.18* 
./debs/ipmitool-xcat_1.8.18_amd64.deb
./debs/ipmitool-xcat_1.8.18_ppc64el.deb
[root@c910f02c07p02 xcat-dep]# find ./ -name ipmitool*1.8.18*  | xargs sum
26689   279 ./debs/ipmitool-xcat_1.8.18_amd64.deb
24693   259 ./debs/ipmitool-xcat_1.8.18_ppc64el.deb
```

Removing the 1.8.17 version...
```
[root@c910f02c07p02 xcat-dep]# rm -f ./debs/ipmitool-xcat_1.8.17-1_amd64.deb
[root@c910f02c07p02 xcat-dep]# rm -f ./debs/ipmitool-xcat_1.8.17-1_ppc64el.deb
```


@zet809  @chenglch What are these files here?  Can I delete them? 
```
[root@c910f02c07p02 xcat-dep]# find ./ -name ipmitool* 
./debs/ipmitool-xcat_1.8.18_amd64.deb
./debs/ipmitool-xcat_1.8.18_ppc64el.deb
./xcat-dep/pool/main/i/ipmitool-xcat
./xcat-dep/pool/main/i/ipmitool-xcat/ipmitool-xcat_1.8.11-1_amd64.deb <---
./xcat-dep/pool/main/i/ipmitool-xcat/ipmitool-xcat_1.8.11-1_ppc64el.deb <---
```

**Comment by neo954** [comment]
@vhu, I believe for the Debian repository, the files under directory `./xcat-dep/pool/main/i/ipmitool-xcat/` need to be updated as well. Or maybe they are hard links to files under directory `./debs/`.

**Comment by chenglch** [comment]
I do not know the purpose of these files. Seems we haven't updated them for a long time.

**Comment by zet809** [comment]
I think the ./xcat-dep directory in the same directory with ./debs is generated wrongly and shall be removed. You can see the xcat-genesis-base are also very old.
# find ./xcat-dep -name "xcat-genesis-*"
./xcat-dep/pool/main/x/xcat-genesis-base-ppc64
./xcat-dep/pool/main/x/xcat-genesis-base-ppc64/xcat-genesis-base-ppc64_2.9-snap201503171145_all.deb
./xcat-dep/pool/main/x/xcat-genesis-base-amd64
./xcat-dep/pool/main/x/xcat-genesis-base-amd64/xcat-genesis-base-amd64_2.9-snap201503230106_all.deb

**Comment by whowutwut** [comment]
Ok I'll remove the ipmitool-xcat ones and build xcat-deps. let's see what happens.

**Comment by whowutwut** [comment]
Did someone create those two updated ones? 
```
[root@c910f02c07p02 xcat-dep]# find ./ -name "ipmitool-xcat*" -print
./debs/ipmitool-xcat_1.8.18_amd64.deb
./debs/ipmitool-xcat_1.8.18_ppc64el.deb
./xcat-dep/pool/main/i/ipmitool-xcat
./xcat-dep/pool/main/i/ipmitool-xcat/ipmitool-xcat_1.8.18_amd64.deb
./xcat-dep/pool/main/i/ipmitool-xcat/ipmitool-xcat_1.8.18_ppc64el.deb
```

**Comment by whowutwut** [comment]
Created new xcat-dep off of the 6/15/17 daily build.  

This new xcat-dep is required with the changes merged in #3252 . 

If you see the following problem on your system after installing the development build: 
```
console: SSL negotiation failed
70367486123328:error:14094418:SSL routines:SSL3_READ_BYTES:tlsv1 alert unknown ca:s3_pkt.c:1259:SSL alert number 48
```
Upgrade conserver and run `makeconservercf` 
```
# rpm -Uhv conserver-xcat-8.2.1-1.ppc64le.rpm
# makeconservercf
```

### LINUX

Under: `/media/xcat/build/xcat2_autobuild_daily_builds/20170616.0615/linux_rpm/devel/src/xcat-core` executed `/builddep.sh UP=1 BUILDALL=1`

Result: http://xcat.org/files/xcat/xcat-dep/2.x_Linux/xcat-dep-201706161120.tar.bz2

### Debian

Under `/media/xcat/build/xcat2_autobuild_daily_builds/20170616.0615/linux_debian/devel/src/xcat-core/` executed: `./build-ubunturepo -d UP=1 BUILDALL=1`

Result: http://xcat.org/files/xcat/xcat-dep/2.x_Ubuntu/xcat-dep-ubuntu-201706161124.tar.bz2

**Comment by whowutwut** [comment]
One problem here is that we are missing some documentation steps for updating the xcat-deps..   
I can create a pull request for this...  We need the following commands to pick up the `ipmitool-xcat` and `conserver-xcat` packages 

```
yum update "*xCAT*"
yum update "*xcat*" 
```
